File: myblog/src/services/profileService.js
import axios from 'axios'
import defaultAvatar from '@/assets/images/avatar.jpg'
import school1Image from '@/assets/images/school1.jpeg'
import school2Image from '@/assets/images/school2.jpeg'

const PUBLIC_BASE_URL = 'http://localhost:8080/api/public'
const ADMIN_BASE_URL = 'http://localhost:8080/api/admin'

function getAuthHeaders() {
  const token = localStorage.getItem('jwtToken')
  return token ? { Authorization: `Bearer ${token}` } : {}
}

function unwrapAdminResponse(response, fallbackMessage) {
  if (response.data.code !== '0000') {
    throw new Error(response.data.info || fallbackMessage)
  }
  return response.data.data
}

function normalizeAsset(value, fallback) {
  if (!value || value.startsWith('@/assets/')) {
    return fallback
  }
  return value
}

export const DEFAULT_PROFILE = {
  avatar: defaultAvatar,
  name: 'Idealer',
  bio: 'Java开发工程师 | 大模型应用开发工程师 | 大模型算法工程师',
  email: '2755027635@qq.com',
  githubName: 'idealer-liXi',
  githubUrl: 'https://github.com/idealer-liXi',
  location: '辽宁-沈阳',
  introduction: '具备扎实的前端、后端开发能力，熟悉Vue、SpringBoot等主流框架，热爱开源和技术分享。喜欢设计美观、体验流畅的Web应用，追求代码的优雅与高效。乐于团队协作，善于沟通，持续学习新技术。',
  hobbies: ['编程与算法', '开源社区', '技术分享', '音乐与吉他', '美食探索'],
  schools: [
    {
      schoolKey: 'undergraduate',
      image: school1Image,
      name: '安徽理工大学',
      tags: ['省部共建', '中西部高校基础能力建设工程'],
      description: '坐落于安徽省淮南市，是安徽省高等教育振兴计划‘地方特色高水平大学建设’项目立项建设高校。',
      honors: ['国家级大学生创新创业训练计划', '全国大学生数学建模竞赛一等奖']
    },
    {
      schoolKey: 'graduate',
      image: school2Image,
      name: '东北大学',
      tags: ['985', '211', '双一流'],
      description: '坐落于辽宁省沈阳市，是教育部直属的全国重点大学，由教育部、国防科工局、辽宁省、沈阳市共建。',
      honors: ['暂无~~']
    }
  ]
}

function mergeProfile(payload) {
  if (!payload) return DEFAULT_PROFILE

  const schools = DEFAULT_PROFILE.schools.map((school) => {
    const matched = (payload.schools || []).find(item => item.schoolKey === school.schoolKey) || {}
    return {
      ...school,
      honors: Array.isArray(matched.honors) && matched.honors.length > 0 ? matched.honors : school.honors
    }
  })

  return {
    avatar: normalizeAsset(payload.avatar, DEFAULT_PROFILE.avatar),
    name: payload.name || DEFAULT_PROFILE.name,
    bio: payload.bio || DEFAULT_PROFILE.bio,
    email: payload.email || DEFAULT_PROFILE.email,
    githubName: payload.githubName || DEFAULT_PROFILE.githubName,
    githubUrl: payload.githubUrl || DEFAULT_PROFILE.githubUrl,
    location: payload.location || DEFAULT_PROFILE.location,
    introduction: payload.introduction || DEFAULT_PROFILE.introduction,
    hobbies: Array.isArray(payload.hobbies) && payload.hobbies.length > 0 ? payload.hobbies : DEFAULT_PROFILE.hobbies,
    schools
  }
}

export async function getPublicProfile() {
  const response = await axios.get(`${PUBLIC_BASE_URL}/profile`)
  return mergeProfile(response.data)
}

export async function getAdminProfile() {
  const response = await axios.get(`${ADMIN_BASE_URL}/profile`, {
    headers: getAuthHeaders()
  })
  return mergeProfile(unwrapAdminResponse(response, '获取站主介绍失败'))
}

export async function updateProfile(payload) {
  const response = await axios.put(`${ADMIN_BASE_URL}/profile`, payload, {
    headers: { ...getAuthHeaders(), 'Content-Type': 'application/json' }
  })
  return mergeProfile(unwrapAdminResponse(response, '更新站主介绍失败'))
}

File: backend/backend-domain/src/main/java/cn/idealer01/domain/profile/service/ProfileService.java
package cn.idealer01.domain.profile.service;

import cn.idealer01.api.dto.ProfileRequestDTO;
import cn.idealer01.api.dto.ProfileResponseDTO;
import cn.idealer01.domain.profile.adapter.repository.IProfileSchoolHonorsRepository;
import cn.idealer01.domain.profile.adapter.repository.IProfileSettingsRepository;
import cn.idealer01.domain.profile.model.entity.ProfileSchoolHonorsEntity;
import cn.idealer01.domain.profile.model.entity.ProfileSettingsEntity;
import cn.idealer01.types.enums.ResponseCode;
import cn.idealer01.types.exception.AppException;
import org.springframework.stereotype.Service;

import javax.annotation.Resource;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Date;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class ProfileService implements IProfileService {

    private static final String SCHOOL_KEY_UNDERGRADUATE = "undergraduate";
    private static final String SCHOOL_KEY_GRADUATE = "graduate";

    @Resource
    private IProfileSettingsRepository profileSettingsRepository;

    @Resource
    private IProfileSchoolHonorsRepository profileSchoolHonorsRepository;

    @Override
    public ProfileResponseDTO getPublicProfile() {
        return buildProfileResponse();
    }

    @Override
    public ProfileResponseDTO getAdminProfile() {
        return buildProfileResponse();
    }

    @Override
    public ProfileResponseDTO updateProfile(ProfileRequestDTO request) {
        validateRequest(request);

        profileSettingsRepository.saveOrUpdate(ProfileSettingsEntity.builder()
                .avatar(defaultString(request.getAvatar()))
                .name(defaultString(request.getName()))
                .bio(defaultString(request.getBio()))
                .email(defaultString(request.getEmail()))
                .githubName(defaultString(request.getGithubName()))
                .githubUrl(defaultString(request.getGithubUrl()))
                .location(defaultString(request.getLocation()))
                .introduction(defaultString(request.getIntroduction()))
                .hobbies(joinLines(request.getHobbies()))
                .updateTime(new Date())
                .build());

        profileSchoolHonorsRepository.saveOrUpdate(ProfileSchoolHonorsEntity.builder()
                .schoolKey(SCHOOL_KEY_UNDERGRADUATE)
                .honors(joinLines(request.getUndergraduateHonors()))
                .updateTime(new Date())
                .build());

        profileSchoolHonorsRepository.saveOrUpdate(ProfileSchoolHonorsEntity.builder()
                .schoolKey(SCHOOL_KEY_GRADUATE)
                .honors(joinLines(request.getGraduateHonors()))
                .updateTime(new Date())
                .build());

        return buildProfileResponse();
    }

    private ProfileResponseDTO buildProfileResponse() {
        ProfileSettingsEntity settings = profileSettingsRepository.getCurrent();
        List<ProfileSchoolHonorsEntity> honorsEntities = profileSchoolHonorsRepository.findAll();
        ProfileResponseDTO defaults = buildDefaultProfile();

        return ProfileResponseDTO.builder()
                .avatar(hasText(settings == null ? null : settings.getAvatar()) ? settings.getAvatar() : defaults.getAvatar())
                .name(hasText(settings == null ? null : settings.getName()) ? settings.getName() : defaults.getName())
                .bio(hasText(settings == null ? null : settings.getBio()) ? settings.getBio() : defaults.getBio())
                .email(hasText(settings == null ? null : settings.getEmail()) ? settings.getEmail() : defaults.getEmail())
                .githubName(hasText(settings == null ? null : settings.getGithubName()) ? settings.getGithubName() : defaults.getGithubName())
                .githubUrl(hasText(settings == null ? null : settings.getGithubUrl()) ? settings.getGithubUrl() : defaults.getGithubUrl())
                .location(hasText(settings == null ? null : settings.getLocation()) ? settings.getLocation() : defaults.getLocation())
                .introduction(hasText(settings == null ? null : settings.getIntroduction()) ? settings.getIntroduction() : defaults.getIntroduction())
                .hobbies(resolveHobbies(settings, defaults))
                .schools(resolveSchools(honorsEntities, defaults.getSchools()))
                .build();
    }

    private List<String> resolveHobbies(ProfileSettingsEntity settings, ProfileResponseDTO defaults) {
        List<String> hobbies = splitLines(settings == null ? null : settings.getHobbies());
        return hobbies.isEmpty() ? defaults.getHobbies() : hobbies;
    }

    private List<ProfileResponseDTO.SchoolProfileDTO> resolveSchools(List<ProfileSchoolHonorsEntity> honorsEntities,
                                                                     List<ProfileResponseDTO.SchoolProfileDTO> defaultSchools) {
        ProfileResponseDTO.SchoolProfileDTO undergraduate = defaultSchools.get(0);
        ProfileResponseDTO.SchoolProfileDTO graduate = defaultSchools.get(1);

        for (ProfileSchoolHonorsEntity entity : honorsEntities) {
            if (SCHOOL_KEY_UNDERGRADUATE.equals(entity.getSchoolKey())) {
                undergraduate = mergeSchoolHonors(undergraduate, entity.getHonors());
            }
            if (SCHOOL_KEY_GRADUATE.equals(entity.getSchoolKey())) {
                graduate = mergeSchoolHonors(graduate, entity.getHonors());
            }
        }

        return Arrays.asList(undergraduate, graduate);
    }

    private ProfileResponseDTO.SchoolProfileDTO mergeSchoolHonors(ProfileResponseDTO.SchoolProfileDTO base, String honorsText) {
        List<String> honors = splitLines(honorsText);
        return ProfileResponseDTO.SchoolProfileDTO.builder()
                .schoolKey(base.getSchoolKey())
                .name(base.getName())
                .tags(base.getTags())
                .description(base.getDescription())
                .image(base.getImage())
                .honors(honors.isEmpty() ? base.getHonors() : honors)
                .build();
    }

    private void validateRequest(ProfileRequestDTO request) {
        if (request == null) {
            throw new AppException(ResponseCode.ILLEGAL_PARAMETER.getCode(), "站主介绍配置不能为空");
        }
        if (!hasText(request.getName())) {
            throw new AppException(ResponseCode.ILLEGAL_PARAMETER.getCode(), "姓名不能为空");
        }
        if (!hasText(request.getBio())) {
            throw new AppException(ResponseCode.ILLEGAL_PARAMETER.getCode(), "职业描述不能为空");
        }
    }

    private ProfileResponseDTO buildDefaultProfile() {
        return ProfileResponseDTO.builder()
                .avatar("@/assets/images/avatar.jpg")
                .name("Idealer")
                .bio("Java开发工程师 | 大模型应用开发工程师 | 大模型算法工程师")
                .email("2755027635@qq.com")
                .githubName("idealer-liXi")
                .githubUrl("https://github.com/idealer-liXi")
                .location("辽宁-沈阳")
                .introduction("具备扎实的前端、后端开发能力，熟悉Vue、SpringBoot等主流框架，热爱开源和技术分享。喜欢设计美观、体验流畅的Web应用，追求代码的优雅与高效。乐于团队协作，善于沟通，持续学习新技术。")
                .hobbies(Arrays.asList("编程与算法", "开源社区", "技术分享", "音乐与吉他", "美食探索"))
                .schools(Arrays.asList(
                        ProfileResponseDTO.SchoolProfileDTO.builder()
                                .schoolKey(SCHOOL_KEY_UNDERGRADUATE)
                                .name("安徽理工大学")
                                .image("@/assets/images/school1.jpeg")
                                .tags(Arrays.asList("省部共建", "中西部高校基础能力建设工程"))
                                .description("坐落于安徽省淮南市，是安徽省高等教育振兴计划‘地方特色高水平大学建设’项目立项建设高校。")
                                .honors(Arrays.asList("国家级大学生创新创业训练计划", "全国大学生数学建模竞赛一等奖"))
                                .build(),
                        ProfileResponseDTO.SchoolProfileDTO.builder()
                                .schoolKey(SCHOOL_KEY_GRADUATE)
                                .name("东北大学")
                                .image("@/assets/images/school2.jpeg")
                                .tags(Arrays.asList("985", "211", "双一流"))
                                .description("坐落于辽宁省沈阳市，是教育部直属的全国重点大学，由教育部、国防科工局、辽宁省、沈阳市共建。")
                                .honors(Collections.singletonList("暂无~~"))
                                .build()
                ))
                .build();
    }

    private String joinLines(List<String> values) {
        if (values == null) {
            return "";
        }
        return values.stream().map(String::trim).filter(this::hasText).collect(Collectors.joining("\n"));
    }

    private List<String> splitLines(String value) {
        if (!hasText(value)) {
            return new ArrayList<>();
        }
        return Arrays.stream(value.split("\\r?\\n")).map(String::trim).filter(this::hasText).collect(Collectors.toList());
    }

    private String defaultString(String value) {
        return value == null ? "" : value.trim();
    }

    private boolean hasText(String value) {
        return value != null && !value.trim().isEmpty();
    }
}
